src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/HttpBasicFactory.php

```diff
@@ -66,7 +67,7 @@ public function addConfiguration(NodeDefinition $node)
 
     protected function createEntryPoint($container, $id, $config, $defaultEntryPoint)
     {
-        if (null !== $defaultEntryPoint) {
+        if (null !== $defaultEntryPoint && $defaultEntryPoint instanceof BasicAuthenticationEntryPoint) {
```

Comment:
This is broken, `$defaultEntryPoint` is a service id here (string), and we cannot do`findDefinition($defaultEntryPointId)->getClass()` since the definition won't exist at this stage.
I fear we need a compiler pass to fix this...

Reply:
Sorry indeed this is weird. It just works because we never enter in the next statement. The entry point is created even if there is a default matching entry point.